### PR TITLE
stats: Various CSM Observability bug fixes

### PR DESCRIPTION
**Pull request by zasweq**
This PR fixes various bugs I found while interop testing CSM Observability.

1. "service_namespace" -> "service_namespace_name
2. OpenTelemetry component default metrics were not working properly
3. Have a version as a separate call

RELEASE NOTES: N/A

### CODE REVIEW

File: stats/opentelemetry/client_metrics.go [hunk 1]
```diff
@@ -44,12 +44,15 @@ func (h *clientStatsHandler) initializeMetrics() {
 		return
 	}
 
-	meter := h.o.MetricsOptions.MeterProvider.Meter("grpc-go " + grpc.Version)
+	meter := h.o.MetricsOptions.MeterProvider.Meter("grpc-go ", otelmetric.WithInstrumentationVersion(grpc.Version))
 	if meter == nil {
 		return
 	}
 
-	setOfMetrics := h.o.MetricsOptions.Metrics.metrics
```
Comment (dfawley) [inline stats/opentelemetry/client_metrics.go:52]: Nit: can we name this `options`?  1/2-character variable abbreviations should have small scope, not be fields in structs unless there's some very well established convention (e.g. we use `cc` a lot for `ClientConn`).

File: stats/opentelemetry/client_metrics.go [hunk 2]
```diff
@@ -44,12 +44,15 @@ func (h *clientStatsHandler) initializeMetrics() {
 		return
 	}
 
-	meter := h.o.MetricsOptions.MeterProvider.Meter("grpc-go " + grpc.Version)
+	meter := h.o.MetricsOptions.MeterProvider.Meter("grpc-go ", otelmetric.WithInstrumentationVersion(grpc.Version))
```
Comment (dfawley) [inline stats/opentelemetry/client_metrics.go:47]: Delete this space in the string and the one below please.

File: stats/opentelemetry/server_metrics.go [hunk 3]
```diff
@@ -44,11 +44,14 @@ func (h *serverStatsHandler) initializeMetrics() {
...
-	meter := h.o.MetricsOptions.MeterProvider.Meter("grpc-go " + grpc.Version)
+	meter := h.o.MetricsOptions.MeterProvider.Meter("grpc-go ", otelmetric.WithInstrumentationVersion(grpc.Version))
 	if meter == nil {
 		return
 	}
-	setOfMetrics := h.o.MetricsOptions.Metrics.metrics
+	setOfMetrics := DefaultMetrics.metrics
+	if h.o.MetricsOptions.Metrics != nil {
+		setOfMetrics = h.o.MetricsOptions.Metrics.metrics
+	}
```
Comment (dfawley) [inline stats/opentelemetry/server_metrics.go:54]: Go style prefers:

```go
setOfMetrics := h.o.MetricsOptions.Metrics
if setOfMetrics == nil {
	setOfMetrics = DefaultMetrics.metrics
```

File: stats/opentelemetry/client_metrics.go (hunk 2 above)
Comment (zasweq) [reply stats/opentelemetry/client_metrics.go:47]: Done.

**Review by dfawley — APPROVED** [review]

**Pull request merged**